apps/src/applab/applab.js

```diff
@@ -460,6 +464,15 @@ Applab.initReadonly = function(config) {
   studioApp.initReadonly(config);
 };
 
+function extendHandleClearPuzzle() {
+  var orig = studioApp.handleClearPuzzle.bind(studioApp);
+  studioApp.handleClearPuzzle = function (config) {
+    orig(config);
+    Applab.setLevelHtml(config.level.startHtml || '');
+    studioApp.resetButtonClick();
+  };
+}
```

Comment:
This feels a little bit gross. If we had true OO + inheritance, this would instead be something like calling clearPuzzle on the superclass, then calling our app-specific code. I'm not sure if there's a cleaner way of representing this.